src/unix/luv/lwt_luv.ml

```diff
@@ -0,0 +1,42 @@
+let unix_fd_to_fd : Unix.file_descr -> int = Obj.magic
```

Comment:
This will break on Windows unfortunately, where `Unix.file_descr` is a struct. See https://github.com/ocaml/ocaml/pull/1990 for some ongoing work by @dra27 to expose this function more systematically. It may be worth picking out C binding from that PR directly into Lwt to facilitate the conversion.

Reply:
Alternatively, if I understand the PR correctly, it would be disable this engine for Windows and bat the issue back to luv. I don't know how libev does this (because I haven't looked 😊), but my instinct is that the API in luv is wrong and shouldn't expose int, but an abstract type along with a `luv_abstract_file_descr_of_unix_file_descr` function which would deal with this. Surely it affects any Windows users of luv?

Reply:
@dra27 Luv exposes an abstract type. I'm not sure what API you are referring to. Luv works on Windows and is tested on it in CI all the time. I understand it is also being used in production on Windows.

As @ulrikstrid already mentioned, and as we discussed earlier in this PR, all this `Obj.magic` and fd casting business is related only to getting this integration working on Unix "by any means necessary." Once that's done, we will clean up the Unix implementation and get this PR working the right way on Windows, too.

Reply:
Ah well I just realized you mean the polling API specifically, like [here](http://docs.libuv.org/en/v1.x/poll.html#api). libuv exposes it as an `int` &mdash; I understand on Windows this should be a CRT fd, and I don't know which kinds of files have a CRT fd on Windows (but at least some do). In the long term, we may have to do some PRs to libuv itself to get this to work fully on Windows at the libuv level, and take a look at what is already available in the libuv 2.x branch. In the even longer term, most of Lwt would be rewritten to bypass libuv poll handles and use libuv directly, if we went with a thorough conversion. That would make the issue of any polling problems on the libuv level moot for most purposes.

Reply:
(In general, by the way, the libuv API is somewhat sloppy and inconsistent in how it represents fds across its various modules, but I am sure we can get the right dynamics for getting everything working)

Reply:
I added the code that I got from the luv history so there's no `Obj.magic` here anymore. Hopefully that code will be added back into `Luv` at some point so that we don't have to maintain it here since it might be flaky.